### ISSUE TITLE
Github repository not found

### ISSUE THREAD

**Issue by erlebach** [mnoukhov/gradescope-autograder-template#3]
Hi,

I have followed your instructions closely (as far as I can tell). Here is my first question: 
I am cloning my private repo (which is clone of sample-python-pytest-autograder). Should this repository be set for ssh access for https access? So far, I get failures in both modes. More specifically, when I upload the zip file to Gradescope, my output log file gives: 
```
...
ollecting tomli>=1.0.0
  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting iniconfig
  Downloading iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
Collecting exceptiongroup>=1.0.0rc8
  Downloading exceptiongroup-1.1.3-py3-none-any.whl (14 kB)
Installing collected packages: tomli, pluggy, packaging, iniconfig, exceptiongroup, pytest
Successfully installed exceptiongroup-1.1.3 iniconfig-2.0.0 packaging-23.2 pluggy-1.3.0 pytest-7.4.3 tomli-2.0.1
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
# github.com:22 SSH-2.0-babeld-542480ac
fatal: repository 'fsu-sc/PRIVATE-cs8-s18-labxx-gs.git' does not exist    <<<<<<< ERROR >>>>>>>
No apt-get.sh found in repo
No requirements.txt found in repo
No requirements3.txt found in repo
...
```
My private deploy key was stored in deploy_keys/ (public and private) via running the script `./make_deploy_keys.sh`. 
I would like to get verbose output so I can see the commands executed that generate the output in the output provide by the gradescope interface.

Here are the contents of `./git/config`: 
```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
[remote "origin"]
	url = https://github.com/fsu-sc/PRIVATE-cs8-s18-labxx-gs.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "main"]
	remote = origin
	merge = refs/heads/main
```
Therefore, I added the `-x` option to `setup.sh`, but that did not seem to affect the output provided by Gradescope. 

The `setup.sh` command: 
```
mkdir -p ${BASE_DIR}/${REPO_NAME}
git clone ${GIT_REPO} ${BASE_DIR}/${REPO_NAME}
```
is executed. GIT_REPO has the value: 
```
GIT_REPO=fsu-sc/PRIVATE-cs8-s18-labxx-gs.git
```
and I am able to clone it manually via `git clone $GIT_REPO`. 

My `env.sh` file is: 
```
#!/bin/sh
GIT_REPO=fsu-sc/PRIVATE-cs8-s18-labxx-gs.git
```
and the file `envVars.sh`: 
```
#!/usr/bin/env bash

# Extract just the git repo:
#   https://serverfault.com/questions/417241/extract-repository-name-from-github-url-in-bash

BASE_DIR=/autograder/git-repo
WITHOUT_SUFFIX="${GIT_REPO%.*}"
REPO_NAME="$(basename "${WITHOUT_SUFFIX}")"

echo \$BASE_DIR=$BASE_DIR
echo \$GIT_REPO=$GIT_REPO
echo \$REPO_NAME=$REPO_NAME
```
Somehow my repository is not set up properly from Gradescope's point of view. Ultimately, I need more detailed logs to help my debugging. Any help would be appreciated. Thanks.

   Gordon

**Comment by erlebach** [comment]
I fixed my issue. The GIT_REPO variable has to be the SSH link to the repository. Also, the repository should be read only when inputting the deploy key.

**Comment by mnoukhov** [comment]
yes `GIT_REPO` should be the link to the ssh e.g. `git@github.com:mnoukhov/gradescope-autograder-template.git`

**Comment by mnoukhov** [comment]
And I do recommend making the deploy key read only

**Comment by erlebach** [comment]
Thanks. I had this working a few days ago but then started my actual unit tests, and I ran into problems. Specifically, my grading respository (which contains `grade.sh`) does not appear on the gradescope server, although if I run `run_autograder` on the server, the GitHub files appear. This is strange to say the least. 

What is the best approach to get 15-30 min of Zoom or Team time with an expert so I can get some help? Or is there a user forum for Gradescope that I can questions to? You are very busy, I am sure, so I do not wish to impose. But I am at the end of my rope now. 

Thanks,

   Gordon

**Comment by mnoukhov** [comment]
Hi Gordon, I'm not actually at all associated with gradescope, I just created this repo as part of my creating my own course and iterating it with different TAs. I just tested it by copying my deploy key to `assignment0/deploy_key` and uploading that zip. Assignment 0 works for me, does it work for you as well?

I don't know where you got `grade.sh` from but my repo isn't made for that. Just to help unblock you, feel free to schedule a 15min call with me on Monday or Tuesday after 2pm ET using my email `mnoukhov@gmail.com` by sending a calendar invite with a meeting link.

**Comment by erlebach** [comment]
Hi Michael,

Thanks for the offer. It so happens I got unblocked yesterday once I
understood how to use ssh. I had not realized that I should run
`runautograder` AFTER the ssh. I was then able to debug. I had also not
realized that it was the use of decorators in the tests that were
responsible for creating the json file. Now everything works. I simplified
your implementation for my own workflow, which is to use a repository
template for the students, as opposed to asking the students to load files
with specific names, which for me, makes testing easier.

So there is no longer a need to schedule a call, but I appreciate the
offer.

Cheers,

   Gordon
